pkg/combustion/rpm.go

```diff
@@ -52,49 +55,114 @@ func configureRPMs(ctx *image.Context) ([]string, error) {
 	return []string{script}, nil
 }
 
-func getRPMFileNames(rpmSourceDir string) ([]string, error) {
-	var rpmFileNames []string
+func handleRPMs(ctx *image.Context) (repoName string, pkgToInstall []string, err error) {
+	if isResolutionNeeded(ctx) {
```

Comment:
We should always perform resolution in my opinion; we can provide a way to force-skip if we think that's desirable, e.g. in situations where the customer *knows* that resolution is not required, e.g. RPM's have no dependencies, but I'd much rather us prioritise a workflow where we can guarantee RPM installation success during boot because we've used the initial ISO as the base platform (and RPM database) and we've solved for dependency issues prior to boot, and during the EIB process (where logs are also more easily discovered).

Reply:
Force-skipping is just begging for trouble IMO. I will refactor the code so that we always do resolution at least against our internal SUSE repositories. This will require of the user to always provide a `RegCode` so that we can connect to PackageHub. Always doing resolution will greatly reduce the complexity of the code. Adding this to my "TODO" list.

Reply:
Post GA I’d like for us to also look at OpenSUSE MicroOS and/or Leap as an input image that can be used from a community perspective. So let’s not yet tie validation to the RegCode and let’s think about how we could factor that in later.

Reply:
Yea, but in order to always do RPM resolution we need to make sure that the `RegCode` is there, or if it is not there check for `AdditionalRepos` and fail if both are missing. If we do not do such validations we cannot confidently say that we are always doing resolution, because we open the door for the user to be able to pass a RPM without providing `RegCode`/`AdditionalRepo`, which will inevitably result in a bad/confusing user-experience IMO. 

I do not mind not doing any validation for the `RegCode`/`AdditionalRepos`, but we need to understand what this means and document the needed configurations/scenarios accordingly.

Reply:
As discussed, we always do resolution, without currently doing a validation on whether a `RegCode`/`AdditionalRepos` are provided. I will document the need for this and use-cases in the upcoming documentation PR.